<?xml version="1.0"?>

<launch>
	<!-- vim: set ft=xml noet : -->
	<!-- example launch script for PX4 based FCU's -->

	<arg name="id" default="1" />
	<arg name="drone_ip" default="192.168.31.11" />
	<arg name="drone_port" default="14560" />
	<arg name="mavros_port" default="9010" />
	
	<arg name="fcu_url" default="udp://:$(arg mavros_port)@$(arg drone_ip):$(arg drone_port)" />
	<arg name="gcs_url" default="udp-b://@" />
	<arg name="tgt_system" default="$(arg id)" />
	<arg name="tgt_component" default="1" />
	<arg name="log_output" default="screen" />
	<arg name="fcu_protocol" default="v2.0" />
	<arg name="respawn_mavros" default="false" />

	<group ns="uav$(arg id)">
	<include file="$(find mavros)/launch/node.launch">
		<arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
		<arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

		<arg name="fcu_url" value="$(arg fcu_url)" />
		<arg name="gcs_url" value="$(arg gcs_url)" />
		<arg name="tgt_system" value="$(arg tgt_system)" />
		<arg name="tgt_component" value="$(arg tgt_component)" />
		<arg name="log_output" value="$(arg log_output)" />
		<arg name="fcu_protocol" value="$(arg fcu_protocol)" />
		<arg name="respawn_mavros" default="$(arg respawn_mavros)" />
	</include>
	</group>
</launch>
